modules/lkqdBidAdapter.js

```diff
@@ -36,21 +36,32 @@ function buildRequests(validBidRequests, bidderRequest) {
   for (let i = 0; i < validBidRequests.length; i++) {
     let bidRequest = validBidRequests[i];
 
+    let sizes = [];
     // if width/height not provided to the ad unit for some reason then attempt request with default 640x480 size
-    if (!bidRequest.sizes || !bidRequest.sizes.length) {
+    if ((!bidRequest.sizes || !bidRequest.sizes.length) && (!bidRequest.mediaTypes || !bidRequest.mediaTypes.banner || !bidRequest.mediaTypes.banner.sizes || !bidRequest.mediaTypes.banner.sizes.length)) {
       utils.logWarn('Warning: Could not find valid width/height parameters on the provided adUnit');
-      bidRequest.sizes = [[640, 480]];
+      sizes = [[640, 480]];
     }
 
     // JWPlayer demo page uses sizes: [640,480] instead of sizes: [[640,480]] so need to handle single-layer array as well as nested arrays
-    if (bidRequest.sizes.length === 2 && typeof bidRequest.sizes[0] === 'number' && typeof bidRequest.sizes[1] === 'number') {
-      let adWidth = bidRequest.sizes[0];
-      let adHeight = bidRequest.sizes[1];
-      bidRequest.sizes = [[adWidth, adHeight]];
+    if (bidRequest.sizes) {
+      sizes = bidRequest.sizes;
+      if (bidRequest.sizes.length === 2 && typeof bidRequest.sizes[0] === 'number' && typeof bidRequest.sizes[1] === 'number') {
+        let adWidth = bidRequest.sizes[0];
+        let adHeight = bidRequest.sizes[1];
+        sizes = [[adWidth, adHeight]];
+      }
+    } else if (bidRequest.mediaTypes && bidRequest.mediaTypes.banner && bidRequest.mediaTypes.banner.sizes) {
```

Comment:
Similar to check on line 41. Consider storing `bidRequest.mediaTypes.banner.sizes` in a variable to avoid having to get this value repeatedly.

Same can be done with `bidRequest.sizes` above

Reply:
Fixed